### PR TITLE
lax.top_k: add axis support

### LINKED ISSUE #25867: Add `axis` argument to `lax.top_k`
**Feature request:** Add an `axis` argument to [`lax.top_k`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.top_k.html).

Other similar reduction functions in [`lax`](https://jax.readthedocs.io/en/latest/jax.lax.html) receive an `axis` or `reduction_dimension` argument:
- [`argmax`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.argmax.html)
- [`argmin`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.argmin.html)
- [`approx_max_k`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.approx_max_k.html)
- [`approx_min_k`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.approx_min_k.html)

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #25867

This adds support for an `axis` argument to the `lax.top_k_p` primitive. At the moment, the stable HLO operation only supports top-k along the last axis, so in the lowering rule I transpose the input and output if required. For the default case, no transpose is performed, and we should not see any change in performance.

A benefit of this change is that the primitive is now closed under batching, so the batching rule becomes quite a bit simpler.

### CODE REVIEW

File: jax/_src/lax/lax.py [hunk 1]
```diff
@@ -3332,21 +3332,23 @@ def sort_key_val(keys: Array, values: ArrayLike, dimension: int = -1,
...
     operand: N-dimensional array of non-complex type.
     k: integer specifying the number of top entries.
+    axis: optional positive integer specifying the axis along which to compute
+      the top ``k`` entries. Default (None) indicates the last axis.
 
   Returns:
     A tuple ``(values, indices)`` where
 
     - ``values`` is an array containing the top k values along the last axis.
     - ``indices`` is an array containing the indices corresponding to values.
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/lax/lax.py]: Fixed

**Review by danielsuo — APPROVED** [review]
